pages/scan-results/ScanResults.js

```diff
@@ -192,29 +193,47 @@ class ScanResults extends Component {
     this.setState({ packages })
   }
 
-  handleSortAlphabetic = () => {
+  setParamsAndState = ( sortMode ) => {
     const { packages } = this.state
-    const sortedList = packages.sort((packA, packB) =>
-      packA.name.localeCompare(packB.name),
-    )
+    const query = packages
+        .map(pack => `${pack.name}@${pack.version}`)
+        .join(',')
+    Router.push(`/scan-results?packages=${query}&sortMode=${sortMode}`)
 
-    this.setState({ packages: sortedList, sortMode: 'alphabetic' })
+    this.setState({ sortMode: sortMode })
+  }
+
+  handleSortAlphabetic = () => {
+    this.setParamsAndState('alphabetic')
   }
 
   handleSortSize = () => {
-    const { packages } = this.state
-    const sortedList = packages.sort((packA, packB) => {
-      const packASize = packA.result ? packA.result.gzip : 0
-      const packBSize = packB.result ? packB.result.gzip : 0
+    this.setParamsAndState('size')
+  }
 
-      return packBSize - packASize
-    })
+  sortPackages = () => {
+    const { packages, sortMode } = this.state
+    let sortedList
 
-    this.setState({ packages: sortedList, sortMode: 'size' })
+    if (sortMode === 'size'){
+      sortedList = packages.sort((packA, packB) => {
+        const packASize = packA.result ? packA.result.gzip : 0
+        const packBSize = packB.result ? packB.result.gzip : 0
+
+        return packBSize - packASize
+      })
+    } else {
+      sortedList = packages.sort((packA, packB) =>
+          packA.name.localeCompare(packB.name),
+      )
+    }
+    return sortedList
```

Comment:
Nit: A more idiomatic way to do this would be –

```js
  sortBy = (criteria) => (packageA, packageB) => Number 
```

Make `sortBy` return a compare function that can be plugged into line 251 (before).

So the usage in `render` would become –

```js
  packages
    .sort(this.sortBy(this.state.sortMode))
    .map(...)
```